### PR TITLE
feat(permission): implement new server / channel permission menus

### LINKED ISSUE #432: Clicking on the close DM button doesnt close the dm
me don't know if it is just me or, whenever me click to close the dm, it doesn't do anything, the dm still stay there.

### PR DESCRIPTION
**Pull request by insertish**
(closes #432)

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,2 @@
-VITE_API_URL=https://api.revolt.chat
```
Comment (HigherOrderLogic) [inline .env:1]: Is this a mistake or intentional?

Comment (janderedev) [reply .env:1]: new feature

File: src/components/navigation/items/ConnectionStatus.tsx [hunk 2]
```diff
@@ -20,7 +22,8 @@ export default function ConnectionStatus() {
     } else if (status === ClientStatus.DISCONNECTED) {
         return (
             <Banner>
-                <Text id="app.special.status.disconnected" />
+                <Text id="app.special.status.disconnected" /> <br />
+                <a onClick={() => client.websocket.connect()}>Reconnect</a>
```
Comment (Evilhotdog) [inline src/components/navigation/items/ConnectionStatus.tsx:26]: Also needs to be translatable

File: src/pages/settings/server/Roles.tsx [hunk 3]
```diff
@@ -1,289 +1,223 @@
...
+                                                    name: e.currentTarget.value,
+                                                })
+                                            }
+                                            contrast
+                                        />
+                                    </p>
+                                </section>
+                                <section>
+                                    <Overline type="subtle">
+                                        Role Colour
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:143]: Needs to be multi-language

File: src/pages/settings/server/Roles.tsx [hunk 4]
```diff
@@ -1,289 +1,223 @@
...
+
+                // Delete the role from this server.
+                function deleteRole() {
+                    server.deleteRole(selected);
+                }
+
+                return (
+                    <div>
+                        <SpaceBetween>
+                            <H1>Edit {currentRole.name}</H1>
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:112]: And here

File: src/pages/settings/server/Roles.tsx [hunk 5]
```diff
@@ -1,289 +1,223 @@
...
+                            <Button
+                                palette="secondary"
+                                disabled={isEqual(
+                                    currentRole,
+                                    currentRoleValue,
+                                )}
+                                onClick={save}>
+                                Save
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:120]: And here

File: src/pages/settings/server/Roles.tsx [hunk 6]
```diff
@@ -1,289 +1,223 @@
...
+                                            }
+                                            onChange={(colour) =>
+                                                setValue({ ...value, colour })
+                                            }
+                                        />
+                                    </p>
+                                </section>
+                                <section>
+                                    <Overline type="subtle">
+                                        Role Options
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:159]: And here (please tell me if there's a way to just have a comment apply to multiple specific lines)

File: src/pages/settings/server/Roles.tsx [hunk 7]
```diff
@@ -1,289 +1,223 @@
...
+                                    <p>
+                                        <Checkbox
+                                            checked={
+                                                currentRoleValue.hoist ?? false
+                                            }
+                                            onChange={(hoist) =>
+                                                setValue({ ...value, hoist })
+                                            }
+                                            description="Display this role above others.">
+                                            Hoist Role
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:170]: And here

File: src/pages/settings/server/Roles.tsx [hunk 8]
```diff
@@ -1,289 +1,223 @@
...
+                                        </Checkbox>
+                                    </p>
+                                </section>
+                                <section>
+                                    <Overline type="subtle">
+                                        Role Ranking
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:176]: And here

File: src/pages/settings/server/Roles.tsx [hunk 9]
```diff
@@ -1,289 +1,223 @@
...
+                                                    ),
+                                                })
+                                            }
+                                            contrast
+                                        />
+                                    </p>
+                                </section>
+                            </>
+                        )}
+                        <h1>Edit Permissions</h1>
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:196]: And here

File: src/pages/settings/server/Roles.tsx [hunk 10]
```diff
@@ -1,289 +1,223 @@
...
+                                setValue({
+                                    ...value,
+                                    permissions,
+                                } as RoleOrDefault)
+                            }
+                        />
+                        {selected !== "default" && (
+                            <>
+                                <hr />
+                                <h1>Danger Zone</h1>
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:209]: And here

File: src/pages/settings/server/Roles.tsx [hunk 11]
```diff
@@ -1,289 +1,223 @@
...
+                                <Button
+                                    palette="error"
+                                    compact
+                                    onClick={deleteRole}>
+                                    Delete Role
```
Comment (Evilhotdog) [inline src/pages/settings/server/Roles.tsx:214]: And lastly here

**Review by Evilhotdog — COMMENTED** [review]
Several places need to be able to display multiple languages

**Review by Evilhotdog — APPROVED** [review]
Looks good to me

**Comment by insertish** [comment]
Ready to merge on my end.

**Pull request merged**